src/parse/parse_parameters.ts

```diff
@@ -1,27 +1,87 @@
 import { guessType } from ".";
-import {
+import { indentationOf } from "./utilities";
+import { getFunctionName } from "./get_function_name";
+import { 
     Argument,
     Decorator,
     DocstringParts,
     Exception,
     KeywordArgument,
     Returns,
     Yields,
+    Method,
+    Attribute
 } from "../docstring_parts";
 
+function parseMethod(
+    parameterTokens: string[],
+    body: string[],
+    functionName: string): DocstringParts {
+        return {
+            name: functionName,
+            decorators: parseDecorators(parameterTokens),
+            args: parseArguments(parameterTokens),
+            kwargs: parseKeywordArguments(parameterTokens),
+            returns: parseReturn(parameterTokens, body),
+            yields: parseYields(parameterTokens, body),
+            exceptions: parseExceptions(body),
+            classes: [],
+            methods: [],
+            attributes: []
+        };
+}
+
+function parseClass(
+    parameterTokens: string[],
+    body: string[],
+    functionName: string): DocstringParts {
+        let args = parseArguments(parameterTokens);
+        let kwargs = parseKeywordArguments(parameterTokens);
+        return {
+            name: functionName,
+            decorators: parseDecorators(parameterTokens),
+            args: args,
+            kwargs: kwargs,
+            returns: undefined,
+            yields: undefined,
+            exceptions: [],
+            classes: [],
+            methods: [],
+            attributes: parseAttributes(body, args, kwargs)
+        };
+}
+
+function parseModule(
+    body: string[],
+    functionName: string): DocstringParts {
+        return {
+            name: functionName,
+            decorators: [],
+            args: [],
+            kwargs: [],
+            returns: undefined,
+            yields: undefined,
+            exceptions: [],
+            classes: parseMethods(body, /(?:class)\s/),
```

Comment:
I think these should be split into 2 functions instead of passing a regex into it